hubtools/src/lib.rs

```diff
@@ -711,9 +823,19 @@ impl RawHubrisArchive {
     pub fn write_default_caboose(
         &mut self,
         version: Option<&String>,
+        epoch: Option<&String>,
     ) -> Result<(), Error> {
-        let out = tlvc_text::pack(&self.generate_default_caboose(version)?);
-        self.write_caboose(&out)
+        let out =
+            tlvc_text::pack(&self.generate_default_caboose(version, epoch)?);
+        self.write_caboose(&out)?;
+
+        // Update the epoch value in the ImageHeader to match the caboose.
+        let default_epoch = "0".to_string();
```

Comment:
This code requires that `generate_default_caboose` and the local `unwrap_or` have the same policy for how to handle a missing `epoch`.

It would be cleaner to put that policy one place (here), and have `generate_default_caboose` take a `u32` (instead of an `Option`).